internal/controller/nats_controller.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	"github.com/kyma-project/nats-manager/pkg/provisioner"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logger "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	eventingv1alpha1 "github.com/kyma-project/nats-manager/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const natsFinalizerName = "nats.kyma-project.io/finalizer"
```

Comment:
Shall we be consistent with the domain name? It is mentioned as `eventing.kyma-project.io` somewhere and somewhere as `nats.kyma-project.io`. Lets use `nats.kyma-project.io` everywhere?